crates/sats/src/builtin_type.rs

```diff
@@ -11,43 +11,43 @@ use enum_as_inner::EnumAsInner;
+    /// The `U8` type. Values [`BuiltinValue::U8(v)`] will have this type.
     U8,
-    /// The `I16` type. Values [`BuiltinValue::I16(v)`](crate::BuiltinValue::I16) will have this type.
+    /// The `I16` type. Values [`BuiltinValue::I16(v)`] will have this type.
     I16,
-    /// The `U16` type. Values [`BuiltinValue::U16(v)`](crate::BuiltinValue::U16) will have this type.
+    /// The `U16` type. Values [`BuiltinValue::U16(v)`] will have this type.
     U16,
-    /// The `I32` type. Values [`BuiltinValue::I32(v)`](crate::BuiltinValue::I32) will have this type.
+    /// The `I32` type. Values [`BuiltinValue::I32(v)`] will have this type.
     I32,
-    /// The `U32` type. Values [`BuiltinValue::U32(v)`](crate::BuiltinValue::U32) will have this type.
+    /// The `U32` type. Values [`BuiltinValue::U32(v)`] will have this type.
     U32,
-    /// The `I64` type. Values [`BuiltinValue::I64(v)`](crate::BuiltinValue::I64) will have this type.
+    /// The `I64` type. Values [`BuiltinValue::I64(v)`] will have this type.
     I64,
-    /// The `U64` type. Values [`BuiltinValue::U64(v)`](crate::BuiltinValue::U64) will have this type.
+    /// The `U64` type. Values [`BuiltinValue::U64(v)`] will have this type.
     U64,
-    /// The `I128` type. Values [`BuiltinValue::I128(v)`](crate::BuiltinValue::I128) will have this type.
+    /// The `I128` type. Values [`BuiltinValue::I128(v)`] will have this type.
     I128,
-    /// The `U128` type. Values [`BuiltinValue::U128(v)`](crate::BuiltinValue::U128) will have this type.
+    /// The `U128` type. Values [`BuiltinValue::U128(v)`] will have this type.
     U128,
-    /// The `F32` type. Values [`BuiltinValue::F32(v)`](crate::BuiltinValue::F32) will have this type.
+    /// The `F32` type. Values [`BuiltinValue::F32(v)`] will have this type.
     F32,
-    /// The `F64` type. Values [`BuiltinValue::F64(v)`](crate::BuiltinValue::F64) will have this type.
+    /// The `F64` type. Values [`BuiltinValue::F64(v)`] will have this type.
     F64,
     /// The UTF-8 encoded `String` type.
-    /// Values [`BuiltinValue::String(s)`](crate::BuiltinValue::String) will have this type.
+    /// Values [`BuiltinValue::String(s)`] will have this type.
     ///
     /// This type exists for convenience and because it is easy to just use Rust's `String` (UTF-8)
     /// as opposed to rolling your own equivalent byte-array based UTF-8 encoding.
     String,
     /// The type of array values where elements are of a base type `elem_ty`.
-    /// Values [`BuiltinValue::Array(array)`](crate::BuiltinValue::Array) will have this type.
+    /// Values [`BuiltinValue::Array(array)`] will have this type.
     Array(ArrayType),
     /// The type of map values consisting of a key type `key_ty` and value `ty`.
-    /// Values [`BuiltinValue::Map(map)`](crate::BuiltinValue::Map) will have this type.
+    /// Values [`BuiltinValue::Map(map)`] will have this type.
```

Comment:
Hm so that's a bit disappointing.  `redundant_explicit_links` did fire, but `broken_intra_doc_links` doesn't after rebasing.

Reply:
Note that this PR is in Draft exactly to see how helpful it would be to run `cargo doc` in CI with pedantic lint settings. And without anybody looking at the output :)